Dockerfile

```diff
@@ -1,17 +1,39 @@
 # Copyright (c) 2019 Zededa, Inc.
 # SPDX-License-Identifier: Apache-2.0
-FROM debian:9.11
+FROM debian:9.11 as build
 RUN apt update && \ 
-    apt install -y g++ make libprotobuf-dev \
+    apt install -y bash libprotobuf-dev \
                    libprotoc-dev protobuf-compiler \
-                   cmake libssl-dev libcurl4-openssl-dev uuid-dev
+                   libssl-dev libcurl4-openssl-dev uuid-dev
+
+# To build iot-identity-service
+RUN apt install -y \
+    curl gcc g++ git jq make pkg-config cmake \
+    libclang1 libssl-dev llvm-dev
+
+# Install Rust & Cargo
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | bash -s -- -y
+ENV PATH="$PATH:/root/.cargo/bin"
+RUN cargo install bindgen --version '^0.54' && \
+    cargo install cbindgen --version '^0.15'
+
 ADD . /
 WORKDIR /eve-tools
 RUN cp proto/api.proto . && \
     protoc --cpp_out=cpp api.proto && \
     make clean && make all && make install
 
-WORKDIR /azure-on-eve
-RUN mkdir build && cd build && \
-    cmake -Drun_unittests=OFF -DUSE_TEST_TPM_INTERFACE_IN_MEM=OFF -DBUILD_SHARED=ON -Duse_cppunittest=OFF .. && \
-    cmake --build .
+RUN git clone https://github.com/Azure/iot-identity-service.git
+
+# This diff iot-identity-service.diff is tied to commit id
+# 15f59c8bd33b1fd8581a74ae6e5ea145c8cb1b9b of iot-identity-service
+RUN cp /iot-identity-service.diff iot-identity-service/ && \
+    cd iot-identity-service && \
+    git reset --hard 15f59c8bd33b1fd8581a74ae6e5ea145c8cb1b9b && \
```

Comment:
Can you not specify this hash in the git clone?

In any case, it would make sense to define it up front so that it is more visible that we use that particular version.

Reply:
I tried looking for option to specifiy the commit hash along with clone command. Could not find any.
Will make the hash an ARG above.

Reply:
done